lib/philomena/filters/query.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Philomena.Filters.Query do
+  alias Philomena.Search.Parser
+
+  defp user_id_transform(_ctx, data) do
```

Comment:
Neither `user_id_transform` nor `creator_transform` here should be necessary due to the controller level filtering. The filter applied at the controller level ensures that either the returned filter is public or that it belongs to the user.